### DISCUSSION TITLE
How to get data for Email Templates and PDF generation?

### DISCUSSION THREAD

**Discussion by bfeddes** [YetiForceCompany/YetiForceCRM#16854] [How to]
So currently I am creating a Project PDF template for the company I work for. I am able to grab data from the Project by typing:
$ (record :  projectname) $ to get the project's name printed on the exported PDF
$ (record : status) $ to get the status of the project
etc...

However, I also want to be able to grab data from Project Tasks. I saw in some other PDF templates provided by Yetiforce, there are other options for parsing such as $ (relatedRecord : XXXX) $ and more. However, if I were to want to grab a specific field from a module other than Projects, in this case, Project Tasks, how would I do that? Can anyone help? For example, I want to be able to grab the Project Task's name, status, etc, like I did for the Project as was shown above.

Thank you in advance to all who help!

**Comment by apcloic** [comment]
Hi @bfeddes 
Try this #15942

**Reply by bfeddes** [reply]
I am trying to get data from Project Tasks to combine with data from Projects to print out all in one PDF. Here is how my formatting looks right now: (note it does not work. The module I am getting other data from is from the Projects module, but I am also trying to get data from the Project Tasks module in the same PDF, which is where my troubles [lie.) 
$(relatedRecord : Project tasks|projecttaskname|description)$ I am trying to get the name of the project task, as well as the description of it.

**Reply by apcloic** [reply]
@bfeddes 
Hi, module name is **ProjectTask** not Projecttasks

I think something like this should work :

``$(relatedRecordsList : ProjectTask|projecttaskname,description|||5)$``

Regards,

**Comment by rskrzypczak** [comment]
Here you have some examples:
https://github.com/YetiForceCompany/YetiForceCRM/blob/c3f7c05d85fe57b2b3b8b740042ca9f8bff4b0f4/app/TextParser.php#L25-L46
